sql/core/src/test/scala/org/apache/spark/sql/CsvFunctionsSuite.scala

```diff
@@ -768,4 +768,32 @@ class CsvFunctionsSuite extends QueryTest with SharedSparkSession {
       context = ExpectedContext(fragment = "to_csv", getCurrentClassCallSitePattern)
     )
   }
+
+  test("SPARK-47497: to_csv support the data of nested structure as pretty strings") {
+    // The item of the Array is a Map
+    val rows = new java.util.ArrayList[Row]()
+    rows.add(Row(1L, Row(2L, "Alice",
+      Array(Map("math" -> 100L, "english" -> 200L, "science" -> null),
+        Map("math" -> 300L, "english" -> 400L, "science" -> 500L)))))
+
+    val valueSchema = StructType(Seq(
+      StructField("age", LongType),
+      StructField("name", StringType),
+      StructField("scores", ArrayType(MapType(StringType, LongType)))))
+    val schema = StructType(Seq(
+      StructField("key", LongType),
+      StructField("value", valueSchema)))
+
+    val df = spark.createDataFrame(rows, schema)
+    val actual1 = df.select(to_csv($"value"))
+    checkAnswer(actual1, Row("2,Alice," +
+      "\"[{math -> 100, english -> 200, science ->}, " +
```

Comment:
Are we really going to settle this down as a formal behaviour? Is this upported in Spark Connect with Scala client?

Reply:
Yes, this be supported in Spark Connect with Scala client, double-check as follows:
<img width="829" alt="image" src="https://github.com/apache/spark/assets/15246973/f446090d-48c0-487a-a960-90b2d720b3b5">

Reply:
What about Spark Connect with Python?

Reply:
Ah, okay it's fixed in https://github.com/apache/spark/pull/45657